src/main/java/vazkii/quark/tweaks/feature/DispensersDyeBlocks.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * This class was created by <ichttt>. It's distributed as
+ * part of the Quark Mod. Get the Source Code in github:
+ * https://github.com/Vazkii/Quark
+ *
+ * Quark is Open Source and distributed under the
+ * CC-BY-NC-SA 3.0 License: https://creativecommons.org/licenses/by-nc-sa/3.0/deed.en_GB
+ *
+ * File Created @ [16/03/2017, 16:32:53 (CET)]
+ */
+package vazkii.quark.tweaks.feature;
+
+import net.minecraft.block.*;
+import net.minecraft.dispenser.IBlockSource;
+import net.minecraft.init.Bootstrap;
+import net.minecraft.init.Items;
+import net.minecraft.item.EnumDyeColor;
+import net.minecraft.item.ItemDye;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.tileentity.TileEntityShulkerBox;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import net.minecraftforge.fml.common.event.FMLInitializationEvent;
+import vazkii.quark.base.module.Feature;
+
+import javax.annotation.Nonnull;
+
+public class DispensersDyeBlocks extends Feature {
+    @Override
+    public void init(FMLInitializationEvent event) {
+        BlockDispenser.DISPENSE_BEHAVIOR_REGISTRY.putObject(Items.DYE, new DyeBehaviour());
+    }
+
+    private static class DyeBehaviour extends Bootstrap.BehaviorDispenseOptional {
+
+        @Override
+        @Nonnull
+        protected ItemStack dispenseStack(IBlockSource source, ItemStack stack) {
+            BlockPos pos = source.getBlockPos().offset(source.getBlockState().getValue(BlockDispenser.FACING));
+            World world = source.getWorld();
+            Block block = world.getBlockState(pos).getBlock();
+            EnumDyeColor color = EnumDyeColor.byDyeDamage(stack.getMetadata());
+            this.successful = true;
```

Comment:
It's because I extend BehaviorDispenseOptional which uses this to determine the sound. It is only set to false in when bonemeal fails to apply (the default minecraft behavour)